docs/cue/reference/components.cue

```diff
@@ -683,6 +692,52 @@ components: {
 				}
 			}
 
+			_proxy: {
+				common:      false
+				description: "Configures an HTTP(S) proxy for Vector to use."
+				required:    false
+				type: object: options: {
+					enabled: {
+						common:      false
+						description: "If false the proxy will be disabled."
+						required:    false
+						type: bool: default: true
+					}
+					http: {
+						common:      false
+						description: "The URL to proxy HTTP requests through."
+						required:    false
+						type: string: {
+							default: null
+							examples: ["http://foo.bar:3128"]
+							syntax: "literal"
+						}
+					}
+					https: {
+						common:      false
+						description: "The URL to proxy HTTPS requests through."
+						required:    false
+						type: string: {
+							default: null
+							examples: ["http://foo.bar:3128"]
+							syntax: "literal"
+						}
+					}
+					no_proxy: {
+						common:      false
+						description: "List of hosts to avoid proxying."
```

Comment:
I think this should make explicit that this is not just a list of exact host names but that patterns are also allowed, and what form the patterns may take. In the examples, I see both what appear to be suffix matches and wildcards, so that should be documented.

Reply:
Mmm, yeah, I missed this with my suggestion to update the env var description https://github.com/timberio/vector/pull/8205#discussion_r679183376. I think we can just copy that here (or share it).